ui/src/app/App.test.tsx

```diff
@@ -152,19 +156,36 @@ describe("App", () => {
         </MemoryRouter>
       </Provider>
     );
+
     expect(
       store
         .getActions()
-        .filter((action) => action.type === "status/checkAuthenticated").length
+        .filter(
+          (action) => action.type === statusActions.checkAuthenticated().type
+        ).length
     ).toBe(1);
+  });
+
+  it("fetches the auth details again when logging out", () => {
+    // Mock the user being previously authenticated, and currently unauthenticated
+    // i.e. they've logged out.
+    jest.spyOn(reactComponentHooks, "usePrevious").mockReturnValue(true);
     state.status.authenticated = false;
-    act(() => {
-      store.dispatch(statusActions.logout());
-    });
```

Comment:
All of the broken tests used this `store.dispatch` method, which is a bit of an odd pattern. IMO we shouldn't be doing this in our unit tests since we end up testing:
- actions are dispatched to the store correctly
- state is reduced correctly
- component is updated with new state

which is probably more suited for an integration test. Anyway I've rewritten the tests that use this with mocked responses.